cmd/cstorvolumeclaim/cstorvolumeclaim.go

```diff
@@ -70,75 +81,128 @@ var (
+	}
+}
+
+// getCVRAnnotations get the annotations for cstorvolumereplica
+func getCVRAnnotations(pool *apis.CStorPool) map[string]string {
+	return map[string]string{
+		"cstorpool.openebs.io/hostname": pool.Labels["kubernetes.io/hostname"],
+	}
+}
+
+// getCVRFinalizer get the finalizer for cstorvolumereplica
+func getCVRFinalizer() []string {
+	return []string{
+		CStorVolumeReplicaFinalizer,
+	}
+}
+
+// getCVROwnerReference get the ownerReference for cstorvolumereplica
+func getCVROwnerReference(cv *apis.CStorVolume) []metav1.OwnerReference {
+	return []metav1.OwnerReference{
+		*metav1.NewControllerRef(cv,
+			apis.SchemeGroupVersion.WithKind(cvKind)),
+	}
+}
+
+// getCVLabels get the labels for cstorvolume
+func getCVLabels(claim *apis.CStorVolumeClaim) map[string]string {
+	return map[string]string{
+		"openebs.io/target-service":    "cstor-target-svc",
+		"openebs.io/persistent-volume": claim.Name,
+	}
+}
+
+// getCVOwnerReference get the ownerReference for cstorvolume
+func getCVOwnerReference(cvc *apis.CStorVolumeClaim) []metav1.OwnerReference {
+	return []metav1.OwnerReference{
+		*metav1.NewControllerRef(cvc,
+			apis.SchemeGroupVersion.WithKind(cvcKind)),
+	}
+}
+
 // getStorageClass return storageclass object for a given storageClass Name.
 // or error if any.
-func getStorageClass(storageClassName string,
+func getStorageClass(
+	scName string,
 ) (*storagev1.StorageClass, error) {
-	if storageClassName == "" {
-		return nil, errors.New("failed to get storageclass: storageclass missing")
+	if scName == "" {
+		return nil, errors.New("failed to get storageclass: name missing")
 	}
-	scObj, err := sc.NewKubeClient().Get(storageClassName, metav1.GetOptions{})
+	scObj, err := sc.NewKubeClient().Get(scName, metav1.GetOptions{})
 	if err != nil {
 		return nil, errors.Wrapf(
 			err,
 			"failed to get storageclass {%s}",
-			storageClassName,
+			scName,
 		)
 	}
 	return scObj, nil
 }
 
-// getReplicationFactor gets the Replication and Consistency factor from the
-// storageclass
-func getReplicationFactor(scName string,
-) (int, int, error) {
+// getReplicationFactor gets the ReplicationFactor from the from given storageclass
+func getReplicationFactor(
+	class *storagev1.StorageClass,
+) (int, error) {
 
-	scObj, err := getStorageClass(scName)
-	if err != nil {
-		return 0, 0, errors.Wrapf(
-			err,
-			"failed to get storageclass obj {%s}",
-			scName,
-		)
-	}
-	count := scObj.Parameters["replicaCount"]
+	count := class.Parameters[ReplicaCount]
 
 	rfactor, err := strconv.Atoi(count)
 	if err != nil {
-		return 0, 0, errors.Wrapf(
-			err,
-			"failed to convert to int {%s}",
-			count,
-		)
+		return 0, err
 	}
-	return rfactor, (rfactor/2 + 1), nil
+	return rfactor, nil
+	//return rfactor, (rfactor/2 + 1), nil
 
 }
 
-// getFromParameters gets replicaCount and storagePoolClaim from
+// getSPC gets replicaCount and storagePoolClaim from
 // storageclass parameter
-func getFromParameters(scName string,
+func getSPC(
```

Comment:
you don't need to return replicacount here again.. you already have a function for that